### PR TITLE
Hint mode feature

### PR DESCRIPTION
**Pull request by coreylevine**
Merge the hint mode feature into the dev branch.

### CODE REVIEW

File: app/src/main/java/org/liberty/android/fantastischmemo/common/AnyMemoDBOpenHelper.java [hunk 1]
```diff
@@ -143,6 +144,12 @@ private void convertOldDatabase(SQLiteDatabase database) {
     @Override
     public void onUpgrade(SQLiteDatabase database, ConnectionSource connectionSource, int oldVersion, int newVersion) {
         Log.v(TAG, "Old version" + oldVersion + " new version: " + newVersion);
+
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/common/AnyMemoDBOpenHelper.java]: unnecessary blank line

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java [hunk 2]
```diff
@@ -373,9 +436,8 @@ public void setAnswerAudioLocation(String answerAudioLocation) {
 
     public boolean isDefaultColor() {
         return questionTextColor == null &&
-               answerTextColor == null &&
-               questionBackgroundColor == null &&
-               answerBackgroundColor == null &&
-               separatorColor.equals(DEFAULT_SEPARATOR_COLOR);
+                answerTextColor == null &&
+                questionBackgroundColor == null &&
+                answerBackgroundColor == null && separatorColor.equals(DEFAULT_SEPARATOR_COLOR);
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: Why is this on the same line? Why was it changed from the original?

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java [hunk 3]
```diff
@@ -175,13 +178,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 return true;
 
             case R.id.editor_menu_br:
-                if(focusView == questionEdit || focusView ==answerEdit || focusView == noteEdit){
+                if(focusView == questionEdit || focusView ==answerEdit || focusView == noteEdit || focusView == hintEdit){
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java]: just add a space for answerEdit to remain consistent with the other 3

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java [hunk 4]
```diff
@@ -175,13 +178,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
                     addTextToView((EditText)focusView, "<br />");
                 }
                 return true;
 
             case R.id.editor_menu_image:
-                if(focusView == questionEdit || focusView ==answerEdit || focusView == noteEdit){
+                if(focusView == questionEdit || focusView ==answerEdit || focusView == noteEdit || focusView == hintEdit){
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java]: space for ==answerEdit

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java [hunk 5]
```diff
@@ -289,7 +292,10 @@ private void removeAudio(){
         } else if (focusView == answerEdit) {
             currentCard.setAnswer(currentCard.getAnswer().replaceAll("<audio src=.*/>", ""));
             ((EditText)focusView).setText(currentCard.getAnswer());
-        } else {
+        } else if(focusView == hintEdit){
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java]: should have space between if and ( according to AOSP -> if (focus..)

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java [hunk 6]
```diff
@@ -289,7 +292,10 @@ private void removeAudio(){
...
+            ((EditText)focusView).setText(currentCard.getHint());
+        }
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/CardEditor.java]: braces should be on same line as else
ex:
} else {
not
}
else {

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/QACardActivity.java [hunk 7]
```diff
@@ -272,6 +273,7 @@ public void onClick(View v) {
             .setHtmlLinebreakConversion(setting.getHtmlLineBreakConversion())
             .setImageSearchPaths(imageSearchPaths);
 
+
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/QACardActivity.java]: remove this white lane to remain consistent with the rest

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/TwoFieldsCardFragment.java [hunk 8]
```diff
@@ -113,13 +133,21 @@ public void onCreate(Bundle bundle) {
 
         }
 
+        for (int i = 0; i < array3.length;  i++) {
+            field3CardFragmentBuilders[i] = (CardFragment.Builder) array3[i];
+
```
Comment (j-suguma) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/TwoFieldsCardFragment.java]: unnecessary white line

**Comment by j-suguma** [comment]
CI build passed
image

**Comment by rekandiah** [comment]
blahhhhh
blahhhhhhhhhhhh


Good job!

**Pull request merged**